src/Compiler/Microsoft.CodeAnalysis.Razor/src/FilePathComparer.cs

```diff
@@ -1,31 +1,28 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-#nullable disable
-
 using System;
 using System.Runtime.InteropServices;
+using Microsoft.AspNetCore.Razor;
 
 namespace Microsoft.CodeAnalysis.Razor;
 
 internal static class FilePathComparer
 {
-    private static StringComparer _instance;
+    private static StringComparer? _instance;
 
     public static StringComparer Instance
     {
         get
         {
-            if (_instance == null && RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
-            {
-                _instance = StringComparer.Ordinal;
-            }
-            else if (_instance == null)
+            return _instance ?? InterlockedOperations.Initialize(ref _instance, GetComparer());
```

Comment:
I know this is going to be weird coming from me, but I actually don't think the thread safety matters here. It's always setting the same value based on conditions that (I really, really hope) aren't going to be changing on the fly, and even if `StringComparer.Ordinal` wouldn't always be the same value, it actually doesn't matter if `Instance` is referentially equal to itself on every call.

Reply:
That's a fair point. :smile: